### PR TITLE
router-linkのto属性でオブジェクトの使用・HyperText型のurl(string型)をpath(RouteLocationNamedRaw)に

### LINKED ISSUE #314: router-linkのパラメータを第二引数で指定するようにする
今は文字列として自分で組み立ててるけど、第二引数にオブジェクトかなんかで指定するやり方に変える
別issueで、パスに型つけられるとよさそう(Next.jsのtyped routesみたいなやつ)(最近ライブラリが出てた気がする？nuxtのだったかも)

### PR DESCRIPTION
**Pull request by dye8128**
### **User description**
close #314
#334 の続きです、対応完了しました

___

### **PR Type**
enhancement, bug fix

___

### **Description**
- `path` プロパティの型を `string` から `RouteLocationNamedRaw` に変更
- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更
- `useParam` 関数で `undefined` の場合にデフォルト値を設定
- `isActive` 計算プロパティのロジックを修正

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>22 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>PageInfoPanel.vue</strong><dd><code>`path` プロパティの型変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Index/PageInfoPanel.vue

- `path` プロパティの型を `string` から `RouteLocationNamedRaw` に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-dc0f946f78638e56e5b2ecc0dcbd2be21e303c83e421aaee9e9e48191e2063df">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>PageInfoPanels.vue</strong><dd><code>`path` プロパティの値をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Index/PageInfoPanels.vue

- `path` プロパティの値をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-7d3a53834b02898f5b663f7e765c2122c57fd97d8eca30a12f99122fdc0dd05d">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContentHeader.vue</strong><dd><code>`url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Layout/ContentHeader.vue

- `url` プロパティを `path` に変更し、型を `RouteLocationNamedRaw` に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-38b561ad48fddc377b7b4e288f0cbfec7d6786132ed3b39ed091754e0fcb02be">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>NavigationLinksItem.vue</strong><dd><code>`path` プロパティの型変更と `isActive` ロジック修正</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/NavigationBar/NavigationLinksItem.vue

<li><code>path</code> プロパティの型を <code>string</code> から <code>RouteLocationNamedRaw</code> に変更<br> <li> <code>isActive</code> 計算プロパティのロジックを修正<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-cf5b4e6bb3ee8e062372be531948dfa4af7104b823bf5142a76dc68157b9ac64">+3/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ProjectItem.vue</strong><dd><code>`router-link` の `to` 属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Projects/ProjectItem.vue

- `router-link` の `to` 属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-844a39636e9a996623141b9ea32115ddb5552c6beeed673366b54fa43ae93dbd">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Contest.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Contest.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-d558762d359ad5f4ae0424cef5d1ff91c43f5e01633390e791dc37346b1244a6">+5/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestEdit.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestEdit.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-6b345b9b3a019b15846536d31ccad1be7489f3dcc49e2f18453e8701556194be">+9/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestNew.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestNew.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-2480a3520085ba035bf91cec23e66691c4bb8398c16ea8785b978e50a72ba04e">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestTeamEdit.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestTeamEdit.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-fba1e6a0cebaa7278ce05dc4325ed0db0cdafa561086a629740d23d3ae69d2b3">+14/-5</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestTeamNew.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestTeamNew.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-0cf527c2206797a81333cbb51d2070441144d1dea8944380cb84cb4516818a10">+10/-4</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Contests.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Contests.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-00aadddd7146b1fe74e665c6f9e4354eb37da680aa61978bd6d579d54dd70605">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Event.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Event.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-a575e96bf56c7266c57cdd22bc6606d206114e8988880254045b548acdaa9de0">+5/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Events.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Events.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-5cc626e88ad24ec0a84d9308af9c4d9bc27eb56f3274e7b8b81fb5e78cf41556">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Index.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Index.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-75f76a7ed2a0f45f79ae96d58b3530c71c8b88cc3612a84ffd784ba8df2e3cba">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Project.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Project.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-4e8aa41e439d638b3e4fb2ed6b5fc5fd543a6e78b8e5431f82746141b203122a">+9/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ProjectNew.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ProjectNew.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-02e0c08d10c3b4738460539a64f12e91bdfaaec3367a440492d3e31a4946e5da">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Projects.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Projects.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-363eca605e87df7acfca10ba0acb91ccc250494d61cfbb3c58fdafbc6ca9a82f">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>User.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/User.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-8976b1c1352a547bccc2525fe38d57da840cf2c0723eea7580589dd93929dd24">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserAccountEdit.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/UserAccountEdit.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-6dd39e0ccbeb6480292f207d72db2c009a37026942f2c4a8082d94037f5e2494">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserAccountNew.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/UserAccountNew.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-0e1077c6ce1d02539bd752d0b681d55167c2d645ce8deaf4ceae65eb1ba65de9">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserAccounts.vue</strong><dd><code>`header-texts` の `url` プロパティを `path` に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/UserAccounts.vue

- `header-texts` の `url` プロパティを `path` に変更し、オブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-8fa659e3153dfa9e05cfc9887d7bce3d6fb7562066f065335a501f193c5e4c69">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>routeInfo.ts</strong><dd><code>`path` プロパティの型変更とデフォルト値修正</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/lib/routeInfo.ts

<li><code>path</code> プロパティの型を <code>string</code> から <code>RouteLocationNamedRaw</code> に変更<br> <li> デフォルト値の <code>path</code> を空オブジェクトに変更<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-66096bf46b5d020b831b3b67cc268eb072bf8872454bb534ff2d9df7d5321113">+7/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>param.ts</strong><dd><code>`useParam` 関数のデフォルト値設定</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/lib/param.ts

- `useParam` 関数で `undefined` の場合にデフォルト値を設定

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/336/files#diff-7cd0b634b3f6f4addbacef257e34a5a35abfcff587b5483fcc2557efeab1e568">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: src/components/NavigationBar/NavigationLinksItem.vue [hunk 1]
```diff
@@ -16,11 +16,7 @@ const props = defineProps<Props>()
 const routeInfo = useRouteInfo(toRef(props, 'name'))
 
 const isActive = computed(() => {
-  if (props.path === '/') return currentRoute.path === props.path
-  return (
-    currentRoute.path === props.path ||
-    currentRoute.path.startsWith(`${props.path}/`)
-  )
+  return currentRoute.name === props.name
```
Comment (mehm8128) [inline src/components/NavigationBar/NavigationLinksItem.vue:19]: ロジックが変わっちゃってそうです
1. `/`にアクセスしてるときにTopリンクがactiveにならなくなってしまってそうです
2. コンテストの詳細など、コンテスト一覧より下のページを見ているときにContestリンクがactiveにならなくなってしまってそうです(プロフィールやプロジェクトも同様)

File: src/components/NavigationBar/NavigationLinksItem.vue [hunk 2]
```diff
@@ -16,10 +17,10 @@ const props = defineProps<Props>()
 const routeInfo = useRouteInfo(toRef(props, 'name'))
 
 const isActive = computed(() => {
-  if (props.path === '/') return currentRoute.path === props.path
+  const path = routes.find(route => route.name === props.path.name)?.path
   return (
-    currentRoute.path === props.path ||
-    currentRoute.path.startsWith(`${props.path}/`)
+    currentRoute.name === props.path.name ||
+    currentRoute.path.startsWith(`${path}/`)
```
Comment (mehm8128) [inline src/components/NavigationBar/NavigationLinksItem.vue:23]: これでも多分動きはするのですが、`path`が`undefined`だったときに`startsWith('undefined/')`になっちゃってあんまりきれいじゃないので僕が書いたように`undefined`のときは明示的に`false`を返すようにした方がよさそうです(テンプレートリテラル内にstring型以外を入れられないようにするeslintのルールも確かあったはずです)

**Review by mehm8128 — APPROVED** [review]
↑以外は問題なさそうなので、直したらマージしちゃってOKです

**Comment by dye8128** [comment]
確認抜けでした、ごめんなさい...!
訂正＆マージしました！

**Pull request merged**